pkg/teams/teams.go

```diff
@@ -0,0 +1,96 @@
+package teams
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type teamsClient struct {
+	url        string
+	httpClient *httpClient
+}
+
+func New(url, apiToken string) *teamsClient {
+	return &teamsClient{
+		url: url,
+		httpClient: &httpClient{
+			client:   http.DefaultClient,
+			apiToken: apiToken,
+		},
+	}
+}
+
+func (t *teamsClient) IsAuthorized(repo, team string) bool {
+	query := `query ($repoName: String! $teamSlug: Slug! $authorization: RepositoryAuthorization!) {
+       isRepositoryAuthorized(repoName: $repoName, teamSlug: $teamSlug, authorization: $authorization)
+	}`
+
+	vars := map[string]string{
+		"repoName":      repo,
+		"teamSlug":      team,
+		"authorization": "DEPLOY",
+	}
+
+	respBody := struct {
+		Data struct {
+			IsRepositoryAuthorized bool `json:"isRepositoryAuthorized"`
+		} `json:"data"`
+		Errors []map[string]any `json:"errors"`
+	}{}
+
+	if err := t.teamsQuery(context.Background(), query, vars, &respBody); err != nil {
```

Comment:
Should use a passed context, so that these requests time out if the caller disconnects.
This can be abused to starve hookd of resources.

As hookd is public facing (and anyone with a GitHub token can call it and reach this code path) we need to extra be careful with code like this.